### DISCUSSION TITLE
Documentation Ideas

### DISCUSSION THREAD

**Discussion by benkoshy** [benkoshy/pagy#2] [Ideas]
@ddnexus  Please let me know if these ideas are crazy:

A big problem, for new pagy users is to find out, quickly.
- what features are available to me? In other words, what are the "affordances"

All the information is there, right now! But it's a little tricky for **new users**. I propose to make it even easier. 

1. One page which shows VISUALLY (where applicable) what is available with links to documentation.
2. The other is to slightly tweak the API page. 

Here is the proposed workflow.

```mermaid
graph TD    
    C{Do I know what I want?}
    C -->|Yes| CA[Navigate to API documentation via Clicks]
    C -->|Yes| CB[Navigate via Search Bar ]
    C -->|No| E[What tools does pagy offer me?]
    E --> |Visit Visual Affordances| F[Visual Clues]
    E --> |Visit API Affordances| G[List the API Tools]
```

**Comment by ddnexus** [comment]
hmmm... similar to https://mochajs.org/#test-fixture-decision-tree-wizard-thing.

**Reply by benkoshy** [reply]
I don't propose to present a flow chart to folks reading the documentation - except to you as a high level overview of how users might think.

Let me think out loud from the point of view of a NEW user. (I already know the answer, but I'm faced similar problems. Except I know the pagy library much better than a new user):

- in pagy, is there a component that lists the total number of records? Or is there a helper? I think there is. Where will i find that? `
- hmm this looks interesting. https://ddnexus.github.io/pagy/extras/navs#gsc.tab=0 I don't know. How does it look VISUALLY.  not sure? 
- hmm this looks interesting, does it solve my problem? https://ddnexus.github.io/pagy/api/frontend#pagy_navpagy- 
- hmm is this the one? https://ddnexus.github.io/pagy/api/frontend#pagy_navpagy-
- I still can't find it: let me open a new issue: https://github.com/ddnexus/pagy/issues/307


Here is an example which is already done:
![image](https://user-images.githubusercontent.com/15097447/173992205-b16bf1c2-1bb8-4e0d-9527-48ecd6f725a8.png)

In other words, from the point of view of a new user, the problem to solve:

- What tools are available to me? How can I find it quickly?

**Reply by ddnexus** [reply]
> I don't propose to present a flow chart to folks reading the documentation - except to you as a high level overview of how users might think.

I didn't get it the first time 🙃

**Comment by ddnexus** [comment]
I find that a good help to make decisions, however try the new search implemented by retype: it is actually a lot better than the old Google search on the old doc...

I think we should first focus on formatting the old plain markdown files with all the goodies offered by retype... then, when we will finish, we could add new ideas, if they are still needed

**Reply by benkoshy** [reply]
Agreed. I will reformat using retype. these are just ideas / thinking out loud.

**Comment by benkoshy** [comment]
@ddnexus so the visual elements are taken care of.

I still feel though there needs to be some increased transparency on what the extras can do for someone. I'm not sure how that should be though. Any thoughts?

![image](https://user-images.githubusercontent.com/15097447/231621345-07aeb024-e10e-49c3-b187-920031560797.png)

**Comment by ddnexus** [comment]
At this point I have a simpler structure for users:

- Quick start (follow links)
- If you need something more, just search a relevant keyword in the doc

That should get you wherever you need to go, whether it is a How To recipe or a reference to an extra or an API section.

Of course you may want to browse the doc in order to know the whole scenario about what pagy has to offer, but that is mostly for geeks or bloggers.

I don't think we need anything more than what we have now, which is quite sleek IMO.

**Reply by benkoshy** [reply]
> At this point I have a simpler structure for users:
> 
>     Quick start (follow links)
>     If you need something more, just search a relevant keyword in the doc

How will the user know if they need something more? Or how would they know what to search. Typically they'd have to do this operation:

```rb
extras.each do |extra|
   extra.is_this_suitable_for_me?
end
```

Or they'd have to navigate away from the read me page and go here: https://ddnexus.github.io/pagy/categories/extra/

> Of course you may want to browse the doc in order to know the whole scenario about what pagy has to offer, but that is mostly for geeks or bloggers.

I strongly feel it's missing that element of peering from the readme itself into what it does - for everyone including nerds / bloggers. When i get a chance i want to try to see if i can improve it?

**Comment by ddnexus** [comment]
> How will the user know if they need something more? Or how would they know what to search.

Really?

> I strongly feel it's missing that element of peering from the readme itself into what it does

IMO the README has already all the pointers to all the possible info.They are probably too many, and that's why most of the content is collapsed.

IMO adding more stuff would not help.
